tidy3d/components/medium.py

```diff
@@ -2551,8 +2551,11 @@ def _derivative_field_cmp(
         eps_data: PermittivityDataset,
         dim: str,
     ) -> np.ndarray:
-        coords_interp = {key: val for key, val in eps_data.coords.items() if len(val) > 1}
-        dims_sum = {dim for dim in eps_data.coords.keys() if dim not in coords_interp}
+        """Compute derivative with respect to the ``dim`` components within the custom medium."""
+
+        coords_interp = {key: eps_data.coords[key] for key in "xyz"}
+        coords_interp = {key: val for key, val in coords_interp.items() if len(val) > 1}
+        dims_sum = [dim for dim in "xyz" if dim not in coords_interp]
```

Comment:
Not related to this PR but I think we are using this sort of code quite often, maybe we should have some utilities for this.

Reply:
hm, you mean like an interp that ignores dimensions with 1 coordinate?

Reply:
this looks almost like what I need, but I still need to pass the coordinates of the points. it seems to me that spatial interp doesn't accept any coordinates?

Reply:
It's a method of our own `Coords` class. It does feel a bit weird that we can only use it for spatial interp though.

Reply:
sounds good